orchestrator/careplanservice/handle_getquestionnaireresponse.go

```diff
@@ -0,0 +1,30 @@
+package careplanservice
+
+import (
+	"context"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+)
+
+// handleGetQuestionnaire fetches the requested Questionnaire and validates if the requester is authenticated
+func (s *Service) handleGetQuestionnaireResponse(ctx context.Context, id string, headers *fhirclient.Headers) (*fhir.QuestionnaireResponse, error) {
+	// Verify requester is authenticated
+	_, err := auth.PrincipalFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var questionnaireResponse fhir.QuestionnaireResponse
+	err = s.fhirClient.Read("QuestionnaireResponse/"+id, &questionnaireResponse, fhirclient.ResponseHeaders(headers))
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.handleTaskBasedResourceAuth(ctx, headers, questionnaireResponse.BasedOn, "QuestionnaireResponse")
```

Comment:
can you think of situations where allowing updates to `BasedOn` would be a vulnerability?

Reply:
I can't think of situations where we would want to allow updates to BasedOn in the first place, shouldn't this be set during creation and left as immutable?